### PR TITLE
Add wire_order option to all circuit drawers to allow defining the order of the wires

### LINKED ISSUE #8063: arbitrary order in circuit drawers
### What should we add?

Circuit drawers currently have `reverse_bits`, as an option to modify the order in which wires are represented. @ajavadia suggests a generalized version of that feature: `wire_order`. This parameter might define the order of the wires `draw(..., wire_order=[3,0,2,1])` and can eventually replace `reverse_bits` with `draw(..., wire_order='reverse'`.

### PR DESCRIPTION
**Pull request by enavarro51**
### Summary

Fixes #8063

### Details and comments

This PR adds the `wire_order` kwarg to `circuit_drawer()` and `QuantumCircuit.draw()`. It's entered as a list of bits, e.g.,
```
qr = QuantumRegister(4, "q")
cr = ClassicalRegister(4, "c")
cr2 = ClassicalRegister(2, "ca")
circuit = QuantumCircuit(qr, cr, cr2)
circuit.h(0)
circuit.h(3)
circuit.x(1)
circuit.x(3).c_if(cr, 10)
circuit.draw('mpl', cregbundle=False, wire_order=[2, 1, 3, 0, 6, 8, 9, 5, 4, 7])
```
image

The `wire_order` list must contain an entry for every bit, and the bits are numbered from 0 starting with the first qubit and continuing through the qubits and clbits.

The `reverse_bits` option works as it did before, but it now uses the same code that `wire_order` uses.

### CODE REVIEW

**Review by 1ucian0 — CHANGES_REQUESTED** [review]
Looks good, thanks!

Some comments here and there. Also, considerting extending the tests for checking the warnings and exceptions discussed in https://github.com/Qiskit/qiskit-terra/issues/8063#issuecomment-1141991937

File: qiskit/visualization/utils.py [hunk 1]
```diff
@@ -281,14 +276,13 @@ def get_wire_label(drawer, register, index, layout=None, cregbundle=True):
     return wire_label
 
 
-def get_condition_label_val(condition, circuit, cregbundle, reverse_bits):
+def get_condition_label_val(condition, circuit, cregbundle):
```
[the remark this answers is not in the document]
Comment (enavarro51) [reply qiskit/visualization/utils.py]: Done in https://github.com/Qiskit/qiskit-terra/pull/8173/commits/a68cef77321b09ed37f93bba31aa03d6547ec8f8.

File: qiskit/visualization/circuit_visualization.py [hunk 2]
```diff
@@ -297,6 +327,9 @@ def _text_circuit_drawer(
             Default: ``True``.
         encoding (str): Optional. Sets the encoding preference of the output.
             Default: ``sys.stdout.encoding``.
+        wire_order (list): optional. A list of integers used to reorder the display
```
[the remark this answers is not in the document]
Comment (enavarro51) [reply qiskit/visualization/circuit_visualization.py]: Done in https://github.com/Qiskit/qiskit-terra/pull/8173/commits/a68cef77321b09ed37f93bba31aa03d6547ec8f8.

**Review by 1ucian0 — APPROVED** [review]
Thanks!

**Pull request merged**